src/main/java/org/jenkinsci/plugins/durabletask/BourneShellScript.java

```diff
@@ -56,6 +60,17 @@ public String getScript() {
         return script;
     }
 
+    /**
+     * Set of workspaces which we have already run a process in.
+     * Copying output from the controller process consumes a Java thread, so we want to avoid it generally.
+     * But we do it the first time we run a process in a new workspace, to assist in diagnosis.
+     * (For example, if we are unable to write to it due to permissions, we want to see that error message.)
+     * Ideally we would display output the first time a given {@link Launcher} was used in that workspace,
```

Comment:
Note that `docker.image("IMAGE").inside {sh 'echo hello'}` does _not_ print this diagnostic when the workspace is not writable inside the container, since `docker inspect -f . IMAGE` had previously been run outside the container in the same workspace. Perhaps fixable by running the diagnostics for the first _several_ processes in this workspace, etc.